lib/publish.js

```diff
@@ -28,7 +28,25 @@ export default async (pluginConfig, context) => {
   const encodedRepoId = encodeURIComponent(repoId);
   const encodedGitTag = encodeURIComponent(gitTag);
   const encodedVersion = encodeURIComponent(version);
-  const apiOptions = { headers: { "PRIVATE-TOKEN": gitlabToken } };
+  const apiOptions = {
+    headers: {
+      "PRIVATE-TOKEN": gitlabToken,
+    },
+    hooks: {
+      beforeError: [
```

Comment:
As suggested in https://github.com/sindresorhus/got/blob/88e623a0d8140e02eef44d784f8d0327118548bc/documentation/9-hooks.md#beforeerror